File: src/main/scala/com/martomate/tripaint/view/MenuBarAction.scala
package com.martomate.tripaint.view

import com.martomate.tripaint.view.gui.UIAction
import javafx.event.{ActionEvent, EventHandler}
import scalafx.scene.control.{Button, MenuItem, Tooltip}
import scalafx.scene.image.ImageView
import scalafx.scene.input.KeyCombination

case class MenuBarAction(
    text: String,
    imagePath: String = null,
    accelerator: KeyCombination = null,
    action: UIAction = null
)

File: ui/src/main/kotlin/tripaint/view/MenuBarAction.kt
package tripaint.view

import javafx.scene.input.KeyCombination
import tripaint.view.gui.UIAction

data class MenuBarAction(
    val text: String,
    val imagePath: String? = null,
    val accelerator: KeyCombination? = null,
    val action: UIAction? = null
)
